aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -320,5 +322,35 @@ Tensor& ne_out(Tensor& result, const Tensor& self, Scalar other) { return compar
 Tensor ne(const Tensor& self, Scalar other) { return comparison_op(self, other, ne_stub); }
 Tensor& ne_(Tensor& self, Scalar other) { return comparison_op_(self, other, ne_stub); }
 
+Tensor& max_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  auto iter = TensorIterator::binary_op(result, self, other,
+                                        /*check_mem_overlap=*/true);
+  max2_stub(iter.device_type(), iter);
+  TORCH_INTERNAL_ASSERT(result.scalar_type() == iter.output().dtype());
```

Comment:
Can you please clarify, why do you need this check.

Reply:
Right, we shouldn't need this. I removed this line, and replaced it with a check for the input types of both tensors.